component/src/main/java/org/ballerinalang/net/kafka/nativeimpl/KafkaBalServerConnector.java

```diff
@@ -0,0 +1,114 @@
+import org.ballerinalang.net.kafka.impl.KafkaServerConnectorImpl;
+import org.ballerinalang.util.exceptions.BallerinaException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+/**
+ * {@code KafkaBalServerConnector} This is the Kafka implementation for the {@code BallerinaServerConnector} API.
+ */
+@JavaSPIService("org.ballerinalang.connector.api.BallerinaServerConnector")
+public class KafkaBalServerConnector implements BallerinaServerConnector {
+
+    private static final Logger logger = LoggerFactory.getLogger(KafkaBalServerConnector.class);
+
+    private Map<String, KafkaServerConnector> connectorMap = new HashMap<>();
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public List<String> getProtocolPackages() {
+        List<String> protocolPackages = new LinkedList<>();
+        protocolPackages.add(Constants.KAFKA_NATIVE_PACKAGE);
+        return protocolPackages;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void serviceRegistered(Service service) throws BallerinaConnectorException {
+        Annotation kafkaConfig = service.getAnnotationList(Constants.KAFKA_NATIVE_PACKAGE,
+                Constants.ANNOTATION_KAFKA_CONFIGURATION).get(0);
+        if (kafkaConfig == null) {
+            throw new BallerinaException("Error Kafka 'configuration' annotation missing in " + service.getName());
+        }
+
+        Properties configParams = KafkaUtils.processKafkaConsumerConfig(kafkaConfig);
+        String serviceId = service.getName();
+
+        try {
+            KafkaListener kafkaListener = new KafkaListenerImpl(KafkaUtils.extractKafkaResource(service));
+            KafkaServerConnector serverConnector = new KafkaServerConnectorImpl(serviceId,
+                    configParams, kafkaListener);
+            connectorMap.put(serviceId, serverConnector);
+            serverConnector.start();
+        } catch (KafkaConnectorException e) {
+            throw new BallerinaException("Error when starting to listen to the Kafka topic while "
+                    + serviceId + " deployment", e);
+        }
+        if (logger.isDebugEnabled()) {
+            logger.debug("KafkaBalServerConnector registered successfully for service: " + serviceId);
+        }
+    }
+
+//    @Override
+//    public void serviceUnregistered(Service service) throws BallerinaConnectorException {
+//        String serviceId = service.getName();
+//        try {
+//            KafkaServerConnector serverConnector = connectorMap.get(serviceId);
+//            if (null != serverConnector) {
+//                serverConnector.stop();
+//            }
+//        } catch (KafkaConnectorException e) {
+//            throw new BallerinaException("Error while stopping the Kafka server connector related with the service "
+//                    + serviceId, e);
+//        }
+//    }
```

Comment:
Shall we remove this commented code block or add a TODO with message for future reference

Reply:
If something needs to be done properly, do it now, do not postpone it with a TODO! ..

Reply:
+1